doc/on_chain_spec.md

```diff
@@ -669,3 +669,175 @@ that distributing their respective portions of the auction fee
+      Nothing ->
+        startingPrice <= newPrice
+      Just (BidTerms _ oldPrice) ->
+        oldPrice + minimumBidIncrement <= newPrice
+    -- The new standing bid state maintains the same list of approved bidders.
+    guard $ oldApprovedBidders == newApprovedBidders
+```
+
+
+## Signed bidder datums
+
+This section describes an extension to the above spec,
+such that the standing bid is extended with a signed payload by the bidder.
+
+It assumes the `Open Auctions` extension to be implemented
+as it relies on the seller
+for enstablishing the on-chain link between bidder verification key and pubkey hash
+
+
+The idea is to verify on-chain
+that the bidder has signed their new bid,
+and is meant to prevent cases where
+a delegate could submit a false bid for an unaware bidder.
+
+To verify something on-chain,
+we need access to the verification key that produced the signature.
+We already have the bidder `PubKeyHash` available in the `BidTerm`,
+but that can not be used to verify a signature produced by the bidders signing key.
+Unfortunately, it is not possible to even derive the `PubKeyHash` from the verification key on-chain,
+so we will have to carry around both in the datum.
+
+To do this, we must modify `BidTerms` with the bidder vkey,
+and the signature, representing the signed payload.
+
+```haskell
+data BidTerms = BidTerms
+  { bidderPKH :: PubKeyHash
+  -- ^ PubKeyHash of whoever submitted the bid
+  , bidderVK :: BuiltinByteString
+  -- ^ New field. Verification Key of whoever submitted the bid
+  , bidPrice :: Natural
+  -- ^ Which price did the bidder set to buy the auction lot?
+  , bidderSignature :: BuiltinByteString
+  -- ^ New field. Represents the signed payload by the bidder
+  , sellerSignature :: BuiltinByteString
+  -- ^ Represents the signed payload by the seller
+  }
+```
+
+This signature can be generated by the bidder,
+by signing with their signing key,
+a tuple consisting of the `hydraHeadId`, `bidderPKH` and `bidAmount`
+
+```haskell
+generateBidderSignature :: SigningKey PaymentKey -> HeadId -> Natural -> PubKeyHash -> BuiltinByteString
+generateBidderSignature skey headId bidAmount bidderPKH = toBuiltin $ dsign skey $ bidderSignatureMessage headId bidderPKH bidAmount
+
+bidderSignatureMessage :: HeadId -> Natural -> PubKeyHash -> BuiltinByteString
+bidderSignatureMessage headId bidAmount bidderPKH = toByteString headId <> toBysteString bidderPKH
+```
+
+We also need to extend the previous implementation of `sellerSignatureMessage` to also include the bidder verification key in the signature.
+We will use this to verify the association between bidder PKH and VK on-chain.
+
+```haskell
+generateSellerSignature :: SigningKey PaymentKey -> HeadId -> PubKeyHash -> SigningKey VerificationKey ->  BuiltinByteString
+generateSellerSignature skey = toBuiltin . (dsign skey) . sellerSignatureMessage
+
+sellerSignatureMessage :: HeadId -> PubKeyHash -> SigningKey VerificationKey -> BuiltinByteString
```

Comment:
Sorry, but I really disagree on this point. I think we should strive to make this project as simple to understand as possible, it is meant to be a reference implementation for others to pickup.

What is the benefit of making this a separate datatype? All I really need, is to define in a single place where a message is constructed, this way it can be easily changed around if needed, and also shared by on and off chain.

The idea of using something like a phantom type to tag the kind of messages we are building is nice, but it also comes at the cost of making the code more complicated. If we were signing a bunch of different messages all over the place it could make sense, but in our case I don't think it is worth the extra complexity.